README.ops.md

```diff
@@ -121,3 +121,45 @@ Try to use the Terraform upgrade process first. These instructions are captured
 1. Create a new read replica with the exact same name as the old one. This will ensure it has the same DNS name as the old one, and the public DNS alias `ingest-replica1.prd.safecast.cc` will continue to work.
 1. Compare the configuration of the old and new replicas until they match. Ensure that the replica and master are in the same availability zone to minimize any performance or financial costs.
 1. Update [the Terraform configuration](https://github.com/Safecast/infrastructure) to reflect these changes. This repository is private; you may need to ask for access.
+
+## Upgrading ingest Elastic Beanstalk environments
+
+There's two common cases when the Elastic Beanstalk environment for ingest needs to be replaced:
+
+* When upgrading the version of Ruby in use
+* When upgrading the version of Postgres server and client in use.
+
+It's important that the major version of the Postgres client binaries on the application servers matches the version of the Postgres server; otherwise, deployment of the application will fail. Therefore, after upgrading the major version of the Postgres server (e.g. from 11 to 12), upgrade the application servers soon thereafter.
+
+### Upgrading the Ruby version
+
+Upgrade these ingest project files:
+
+* `Gemfile`
+* `.circleci/config.yml`
+* `.ruby-version`
+
+and make sure that all is still working, e.g. by opening a pull request. Then, access the S3 console in AWS and download the following files from the `elasticbeanstalk-us-west-2-985752656544/resources/templates/ingest` folder:
+
+* `dev`
+* `dev-wrk`
+* `prd`
+* `prd-wrk`
+
+In each file, replace the `PlatformArn` line with the new ARN name. Upload the changed files to S3 in the same location.
```

Comment:
You can avoid the explicit s3 bucket stuff here by using `eb config get dev` (edit .elasticbeanstalk/saved_configs/dev-wrk.cfg.yml) `eb config put dev`

The `.elasticbeanstalk/config.global.yml` in each repo points to the beanstalk app which knows the bucket used for configuration. The bucket is always "elasticbeanstalk-us-west-2-985752656544" today, but could be configured on a per-app basis.

Reply:
Thanks, that's a good point. I didn't look too deeply into the CLI while doing this work, and at least one AWS doc tells people to dig into S3 themselves. (You also lose a lot of metadata, e.g. when the config was last updated, when manually changing it in S3).

I was just thinking it would be good to save old configs for rollback, but S3 has versioning built in, doesn't it?

Reply:
yep. Might be good to check that the bucket has versioning turned on, but so long as it is we have history if needed.

Reply:
It looks like it doesn't have versioning turned on. Wonder if anything will break if I change that.